packages/odyssey-react/src/components/Select/Select/index.tsx

```diff
@@ -99,70 +100,72 @@ export type SelectProps = MultipleProps | SingleProps;
  * Often referred to as a "dropdown menu" this input triggers a menu of
  * options a user can select.
  */
-let Select = forwardRefWithStatics<HTMLSelectElement, SelectProps, Statics>(
-  (props, ref) => {
-    const {
-      id,
-      children,
-      disabled = false,
-      name,
-      onChange,
-      required = true,
-      value,
-      error,
-      hint,
-      label,
-      optionalLabel,
-      loadingText = "",
-      noResultsText = "",
-      noChoicesText = "",
-      ...rest
-    } = props;
-
-    const omitProps = useOmit(rest);
-
-    const oid = useOid(id);
-
-    useChoices({ id: oid, value, loadingText, noResultsText, noChoicesText });
-
-    const handleChange = useCallback(
-      (event: ChangeEvent<HTMLSelectElement>) => {
-        onChange?.(event, event.target.value);
-      },
-      [onChange]
-    );
-
-    return (
-      <Field
-        error={error}
-        hint={hint}
-        inputId={oid}
-        label={label}
-        optionalLabel={optionalLabel}
-        required={required}
-      >
-        <div className={styles.outer}>
-          {/* eslint-disable-next-line jsx-a11y/no-onchange */}
-          <select
-            {...omitProps}
-            id={oid}
-            name={name}
-            disabled={disabled}
-            required={required}
-            onChange={handleChange}
-            value={value}
-            ref={ref}
-          >
-            {children}
-          </select>
-          <span className={styles.indicator} role="presentation">
-            <CaretDownIcon />
-          </span>
-        </div>
-      </Field>
-    );
-  }
-);
+let Select = forwardRefWithStatics<
+  ChoicesHTMLSelectElement,
+  SelectProps,
+  Statics
+>((props, ref) => {
+  const {
+    id,
+    children,
+    disabled = false,
+    name,
+    onChange,
+    required = true,
+    value,
+    error,
+    hint,
+    label,
+    optionalLabel,
+    loadingText = "",
+    noResultsText = "",
+    noChoicesText = "",
+    ...rest
+  } = props;
+
+  const omitProps = useOmit(rest);
+
+  const oid = useOid(id);
+
+  useChoices({ id: oid, value, loadingText, noResultsText, noChoicesText });
+
+  const handleChange = useCallback(
```

Comment:
I'm not sure `useCallback` adds anything here?

Reply:
You could argue in the controlled input case its saving *some* memory? Feels probably not worth it to me too with current API. Not sure it's worth changing given component will likely be re-written.